examples/dynamic_allocation.py

```diff
@@ -2,53 +2,118 @@
 # Licensed under the MIT License.
 
 import pyqir
+from pyqir import (
+    BasicBlock,
+    Builder,
+    Context,
+    Function,
+    Linkage,
+    Module,
+    ModuleFlagBehavior,
+)
+
+context = Context()
+mod = Module(context, "dynamic_allocation")
+builder = Builder(context)
 
-mod = pyqir.SimpleModule("dynamic_allocation", num_qubits=0, num_results=0)
-qubit_type = pyqir.qubit_type(mod.context)
-result_type = pyqir.result_type(mod.context)
+# Define module flags
+i1 = pyqir.IntType(context, 1)
+i32 = pyqir.IntType(context, 32)
+
+mod.add_flag(
+    ModuleFlagBehavior.ERROR,
+    "qir_major_version",
+    pyqir.const(i32, 1),
+)
+
+mod.add_flag(
+    ModuleFlagBehavior.MAX,
+    "qir_minor_version",
+    pyqir.const(i32, 0),
+)
+
+mod.add_flag(
+    ModuleFlagBehavior.ERROR,
+    "dynamic_qubit_management",
+    pyqir.const(i1, True),
+)
+
+mod.add_flag(
+    ModuleFlagBehavior.ERROR,
+    "dynamic_result_management",
+    pyqir.const(i1, True),
+)
```

Comment:
Just an idea, but maybe as a parallel to `pyqir.entry_point(...)` that creates a function with all of the necessary attributes, we should have a `pyqir.module(...)` that creates a module with all of the necessary flags? (that SimpleModule would call also)

Reply:
Started in https://github.com/qir-alliance/pyqir/pull/226